lab-said/route/route-toy.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const debug = require('debug')('http:route-toy');
+const Toy = require('../model/toy');
+const storage = require('../lib/storage');
+
+module.exports = function(router) {
+  router.post('/api/toy', (req, res) => {
+    debug('/api/toy POST');
+    try {
+      let newToy = new Toy(req.body.name, req.body.desc, req.body.color, req.body.manif);
+      storage.create('toy', newToy)
+        .then(toy => {
+          res.writeHead(201, {'Content-Type': 'application/json'});
+          res.write(JSON.stringify(toy));
+          res.end();
+        });
+    } catch(e) {
+      console.error(e);
+      res.writeHead(400, {'Content-Type': 'text/plain'});
+      res.write('bad request: could not create a new toy');
+      res.end();
+    }
+  });
+
+  router.put('/api/toy', (req, res) => {
+    debug('/api/toy POST');
+    try {
+      let newToy = new Toy(req.body.name, req.body.desc, req.body.color, req.body.manif);
+      storage.update('toy', newToy)
+        .then(toy => {
+          res.writeHead(201, {'Content-Type': 'application/json'});
+          res.write(JSON.stringify(toy));
+          res.end();
+        });
+    } catch(e) {
+      console.error(e);
+      res.writeHead(400, {'Content-Type': 'text/plain'});
+      res.write('bad request: could not create a new toy');
+      res.end();
+    }
+  });
+
+  router.delete('/api/toy', (req, res) => {
+    debug('/api/toy DELETE');
+    if(req.url.query._id) {
+      storage.delete('toy', req.url.query._id)
+        .then(toy => {
+          res.writeHead(201, {'Content-Type': 'application/json'});
+          res.end();
+        })
+    .catch(err => {
+      console.error(e);
+      res.writeHead(400, {'Content-Type': 'text/plain'});
+      res.write('bad request: could not create a new toy');
+      res.end();
+  });
+
+  router.get('/api/toy', (req, res) => {
+    debug('/api/toy GET');
+    if(req.url.query._id) {
+      storage.fetchOne('toy', req.url.query._id)
+        .then(toy => {
+          res.writeHead(201, {'Content-Type': 'application/json'});
+          res.write(JSON.stringify(toy));
+          res.end();
+        })
+        .catch(err => {
+          console.error(err);
+          res.writeHead(404, {'Content-Type': 'text/plain'});
+          res.write('bad request; could not find record');
+          res.end();
+        })
+      return;
+    }
+
+    res.writeHead(400, {'Content-Type': 'text/plain'});
+    res.write('bad request; item id required to get record');
+    res.end();
+  })
```

Comment:
I can't run the code because there are syntax errors in this file. Probably a missing curly brace somewhere.